app/components/BuyNBTC.tsx

```diff
@@ -1,22 +1,70 @@
 import { Card, CardContent } from "./ui/card";
 import { Button } from "./ui/button";
-import { FormProvider, useForm } from "react-hook-form";
-import { useContext, useCallback, useState } from "react";
+import { FormProvider, useForm, useFormContext } from "react-hook-form";
+import { useContext, useCallback, useState, useEffect } from "react";
 import { WalletContext } from "~/providers/ByieldWalletProvider";
 import { ByieldWallet } from "~/types";
 import { SuiModal } from "./Wallet/SuiWallet/SuiModal";
-import { useSignAndExecuteTransaction, useSuiClient } from "@mysten/dapp-kit";
+import { useCurrentAccount, useSignAndExecuteTransaction, useSuiClient } from "@mysten/dapp-kit";
 import { Transaction } from "@mysten/sui/transactions";
 import { useNetworkVariables } from "~/networkConfig";
-import { mistToSui, suiToMist } from "~/util/util";
-import { pricePerNBTCInSUI } from "~/constant";
+import { createBuyNBTCTxn, mistToSui, suiToMist } from "~/util/util";
+import { BUFFER_BALANCE, PRICE_PER_NBTC_IN_SUI } from "~/constant";
 import { Link } from "@remix-run/react";
 import { ArrowDown, Check, CircleX } from "lucide-react";
 import { useSuiBalance } from "./Wallet/SuiWallet/useSuiBalance";
 import { FormNumericInput } from "./form/FormNumericInput";
 import { classNames } from "~/lib/utils";
+import { Modal } from "./ui/dialog";
+import { NumericFormat } from "react-number-format";
+
+interface FeeProps {
+	fee: number;
+	youReceive: number;
+}
+
+interface BuyNBTCForm {
+	suiAmount: string;
+	amountOfNBTC: string;
+}
+
+function Fee({ fee, youReceive }: FeeProps) {
+	const { formState } = useFormContext<BuyNBTCForm>();
+	const isSUIAmountValid = formState.errors["suiAmount"];
+
+	return (
+		<Card className="p-4 bg-azure-10 rounded-2xl h-20">
+			<CardContent className="flex flex-col justify-between h-full p-0">
+				<div className="flex justify-between">
+					<p className="text-gray-400 text-sm">Estimate Fee</p>
+					<NumericFormat
+						displayType="text"
+						value={fee}
+						suffix=" SUI"
+						allowNegative={false}
+						className="text-sm"
+					/>
+				</div>
+				<div className="flex justify-between">
+					<p className="text-gray-400 text-sm">You Receive</p>
```

Comment:
this should not be there. We should adjust the amount in the nBTC value. Now we have to different nBTC values. Its confusing

Reply:
Done